pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -412,6 +419,8 @@ func (p *testPlanner) Plan() *TestPlan {
 		seed:           p.seed,
 		services:       p.serviceDescriptors(),
 		setup:          setup,
+		rt:             p.rt,
+		ctx:            p.currentContext,
```

Comment:
Would it be better to pass the `testPlanner` to `Generate` instead of adding this information to the `TestPlan`? You need the current context and the roachtest spec to create the plan, but you don't actually need it in the plan itself.